### PR TITLE
[WIP] make op db tests to use multithreaded pg

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89443
* #89442
* #89441
* #89440
* #89439

### CODE REVIEW

File: test/distributed/_tensor/test_dtensor_ops.py [hunk 1]
```diff
@@ -670,11 +671,11 @@ def world_size(self) -> int:
     @ops(dtensor_lagging_op_db, allowed_dtypes=(torch.float,))
     @skipOps("TestDTensorOps", "test_dtensor_op_db", dtensor_fails)
     def test_dtensor_op_db(self, dtype, op):
-        pg_backend = "nccl" if DEVICE_TYPE == "cuda" else "gloo"
-        if pg_backend == "nccl" and torch.cuda.device_count() < self.world_size:
+        # pg_backend = "nccl" if DEVICE_TYPE == "cuda" else "gloo"
```
Comment (yhcharles) [inline test/distributed/_tensor/test_dtensor_ops.py:674]: Any reason to keep this comment?

File: test/distributed/_tensor/test_dtensor_ops.py [hunk 2]
```diff
@@ -670,11 +671,11 @@ def world_size(self) -> int:
...
+        if DEVICE_TYPE == "cuda" and torch.cuda.device_count() < self.world_size:
             sys.exit(TEST_SKIPS[f"multi-gpu-{self.world_size}"].exit_code)
 
-        self.init_pg(backend=pg_backend)
+        # self.init_pg(backend=pg_backend)
```
Comment (yhcharles) [inline test/distributed/_tensor/test_dtensor_ops.py:678]: Ditto

**Review by yhcharles — APPROVED** [review]
LGTM

File: test/distributed/_tensor/test_dtensor_ops.py (hunk 1 above)
Comment (wanchaol) [reply test/distributed/_tensor/test_dtensor_ops.py:674]: this PR is still WIP, might need more fixes to make it to a good shape, let me convert it to a draft

**Pull request closed without merging**